hazelcast/src/main/java/com/hazelcast/jet/pipeline/Sources.java

```diff
@@ -656,21 +604,34 @@ public static <K, V> BatchSource<Entry<K, V>> remoteMap(
      * @since 5.4
      */
     @Nonnull
-    public static <T, K, V> BatchSource<T> remoteMap(
+    public static <K, V, T> BatchSource<T> remoteMap(
             @Nonnull String mapName,
             @Nonnull DataConnectionRef dataConnectionRef,
             @Nonnull Predicate<K, V> predicate,
             @Nonnull Projection<? super Entry<K, V>, ? extends T> projection
     ) {
-        RemoteMapSourceParams<T, K, V> params = RemoteMapSourceParams.<T, K, V>builder(mapName)
-                .withDataConnectionName(dataConnectionRef.getName())
-                .withPredicate(predicate)
-                .withProjection(projection)
+        return Sources.<K, V>remoteMapBuilder(mapName)
+                .dataConnectionName(dataConnectionRef.getName())
+                .predicate(predicate)
+                .projection(projection)
                 .build();
+    }
 
-        ProcessorSupplier processorSupplier = readRemoteMapP(params);
-        return batchFromProcessor("remoteMapSource(" + mapName + ')',
-                ProcessorMetaSupplier.of(processorSupplier));
+    /**
+     * Returns a builder to build a source that fetches entries from a remote
+     * Hazelcast {@code IMap} with the specified name. It provides a fluent API
+     * to build the source using the optional parameters.
+     * <p>
+     * See {@link RemoteMapSourceBuilder} for details on the remote map source.
+     * @param mapName the name of the map
+     * @return builder for remote map source
+     * @param <K> the type of the key in the map
+     * @param <V> the type of the value in the map
+     * @since 5.4
+     */
+    @Nonnull
+    public static <K, V> RemoteMapSourceBuilder<K, V, Entry<K, V>> remoteMapBuilder(String mapName) {
```

Comment:
This method is better renamed to remoteMapSourceBuilder

Reply:
The current convention is not to include the `Source` into the method name - see `files/jmsTopic/jmsQueueBuilder` - depending if you use `Sources.*/Sinks.*` you get a source or sink builder.